ionoscloud/resource_certificate_manager_auto_certificate.go

```diff
@@ -0,0 +1,195 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"slices"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cert"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils/constant"
+)
+
+func resourceCertificateManagerAutoCertificate() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: autoCertificateCreate,
+		ReadContext:   autoCertificateRead,
+		UpdateContext: autoCertificateUpdate,
+		DeleteContext: autoCertificateDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: autoCertificateImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"provider_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The certificate provider used to issue the certificates",
+			},
+			"location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The location of the auto-certificate",
+			},
+			"common_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The common name (DNS) of the certificate to issue. The common name needs to be part of a zone in IONOS Cloud DNS",
+			},
+			"key_algorithm": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The key algorithm used to generate the certificate",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "A certificate name used for management purposes",
+			},
+			"subject_alternative_names": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "Optional additional names to be added to the issued certificate. The additional names needs to be part of a zone in IONOS Cloud DNS",
+			},
+			"last_issued_certificate_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the last certificate that was issued",
+			},
+		},
+		CustomizeDiff: checkAutoCertificateImmutableFields,
+		Timeouts:      &resourceDefaultTimeouts,
+	}
+}
+
+func checkAutoCertificateImmutableFields(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
```

Comment:
same, make them forcenew

Reply:
@cristiGuranIonos done in ec747230649ad96067a92e1d3b11d469d3eda2f1